pkg/accumulator/resaccumulator.go

```diff
@@ -92,28 +105,31 @@ func (ra *ResAccumulator) MergeAccumulator(other *ResAccumulator) (err error) {
 func (ra *ResAccumulator) makeVarReplacementMap() (map[string]string, error) {
 	result := map[string]string{}
 	for _, v := range ra.Vars() {
-		matched := ra.resMap.GetMatchingIds(
-			resid.NewResId(v.ObjRef.GVK(), v.ObjRef.Name).GvknEquals)
-		if len(matched) > 1 {
-			return nil, fmt.Errorf(
-				"found %d resId matches for var %s "+
-					"(unable to disambiguate)",
-				len(matched), v)
-		}
-		if len(matched) == 1 {
-			s, err := ra.resMap[matched[0]].GetFieldValue(v.FieldRef.FieldPath)
-			if err != nil {
-				return nil, fmt.Errorf(
-					"field specified in var '%v' "+
-						"not found in corresponding resource", v)
+		foundVar := false
+		for _, m := range ra.resMap {
+			refVars := m.GetRefVar()
+			for _, varName := range refVars {
```

Comment:
please move this loop to inside the Resource class.

Reply:
done